hack/components.sh

```diff
@@ -65,6 +65,7 @@ apiVersion: apps/v1
 kind: Deployment
 metadata:
   name: mac-controller-manager
+  namespace: system
```

Comment:
We should not hardcode namespace. That should be filled in by kustomize.

Reply:
I'm afraid it's unavoidable in this case. you see, "system" with the added prefix becomes "kubemacpool-system" by kustomize, and only this way the kustomize knows to patch the yamls together. changing it to {{ .Namespace }} ends up with kustomized erroring.

But I will make sure that in the end yaml there would not be any trace of the namespace.

Reply:
I'm think it's not possible. kustomize requires the namespace in order to attach the patch to the appropriate objects. I tried substituting it with  `"{{ .Namespace }}"` or something alike, but it results with the kustomized failing. 

We already have a substitutes every instance of the namespace to the parameterized version,  but I also added another sed command that makes sure that in the end yaml we will have on the parameterized parameter.
is that ok with you?

Reply:
I don't understand why are you changing this. It used to work (correct me if I'm wrong) without this set. Kustomize is configured here https://github.com/kubevirt/cluster-network-addons-operator/blob/a7d8735895c4752ee11f628d431366f7fde5b662/hack/components.sh#L56 to set the namespace.

Reply:
when I use `"{{ .Namespace }}"` ( or just nothing) - I get the following error: 
`Error: no matches for OriginalId apps_v1_Deployment|{{ .Namespace }}|mac-controller-manager; no matches for CurrentId apps_v1_Deployment|{{ .Namespace }}|mac-controller-manager; failed to find unique target for patch apps_v1_Deployment|mac-controller-manager
`
and the output kubemacpol.yaml is empty.

I believe it's because I'm using a newer version of kustomize 3.5.4, and they changed to force specifying an explicit namespace, as mentioned in this [issue](https://github.com/fluxcd/flux/issues/2533)
I'm trying to find a less messier solution..